src/gleam_community/path.gleam

```diff
@@ -73,17 +50,185 @@ pub fn join(path: Path, piece: String) -> Path {
   |> filter_segment_list
 }
 
+/// CONVERSIONS ---------------------------------------------------------------
 pub fn to_string(path: Path) -> String {
   let joined_segments = string.join(path.segments, "/")
 
   case path {
     Absolute(_) -> string.concat(["/", joined_segments])
+
     Relative(_) -> joined_segments
   }
 }
 
+/// 
+pub fn extname(path: Path) -> String {
+  let file_name =
+    path.segments
+    |> list.last()
+    |> result.unwrap("")
+
+  case string.split(file_name, ".") {
+    [_] -> ""
+    file_name_parts ->
+      file_name_parts
+      |> list.last()
+      |> result.unwrap("")
+      |> string.append(".", _)
+  }
+}
+
+// MANIPULATIONS --------------------------------------------------------------
+
+///
+pub fn relative(from from: Path, to to: Path) -> Path {
+  todo
+}
+
+///
 pub fn normalize(messy: String) -> String {
   messy
   |> from_string
   |> to_string
 }
+
+///
+pub fn common(between path_a: Path, and path_b: Path) -> Result(Path, Nil) {
+  case path_a, path_b {
+    Absolute(_), Relative(_) -> Error(Nil)
+    Relative(_), Absolute(_) -> Error(Nil)
+    _, _ -> {
+      let constructor = case path_a {
+        Absolute(_) -> Absolute
+        Relative(_) -> Relative
+      }
+
+      let segments =
+        list.zip(path_a.segments, path_b.segments)
+        |> common_between([])
+
+      case segments {
+        [] if constructor == Absolute -> Ok(constructor([]))
+
+        [] if constructor == Relative -> Error(Nil)
+
+        _ -> Ok(constructor(segments))
+      }
+    }
+  }
+}
+
+// QUERIES --------------------------------------------------------------------
+
+///
+pub fn dirname(path: Path) -> String {
+  case path {
+    Absolute(segments) ->
+      case segments {
+        [] | [_] -> sep()
+        path_parts ->
+          path_parts
+          |> list.take(list.length(path_parts) - 1)
+          |> string.join(sep())
+          |> string.append(sep(), _)
+      }
+
+    Relative(segments) ->
+      case segments {
+        [] | [_] -> "."
+        path_parts ->
+          path_parts
+          |> list.take(list.length(path_parts) - 1)
+          |> string.join(sep())
+          |> string.append("." <> sep(), _)
+      }
+  }
+}
+
+///
+pub fn basename(path: Path) -> String {
+  case result.unwrap(list.last(path.segments), "") {
+    "" ->
+      case path {
+        Absolute(_) -> ""
+        Relative(_) -> "."
+      }
+    base -> base
+  }
+}
+
+///
+pub fn delimiter() -> String {
+  case get_os_type() {
+    "win32" -> ";"
+    "unix" -> ":"
+  }
+}
+
+///
+pub fn sep() -> String {
+  case get_os_type() {
+    "win32" -> "\\"
+    "unix" -> "/"
+  }
+}
+
+// UTILS ----------------------------------------------------------------------
+
+fn filter_segment_list(path: Path) -> Path {
```

Comment:
I know this is my fault, but what an awful name lol

Reply:
It's an internal function so I don't think it's a huge deal but I can try to think of a better name